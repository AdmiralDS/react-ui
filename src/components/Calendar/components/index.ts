export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
