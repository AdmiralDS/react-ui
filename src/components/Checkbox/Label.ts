import styled, { css } from 'styled-components';
import type { CheckboxDimension } from '#src/components/Checkbox/CheckboxDimension';

export const styleTextMixin = css<{ dimension: CheckboxDimension; disabled?: boolean }>`
  font-size: ${({ dimension }) => (dimension === 'm' ? 16 : 14)}px;
  line-height: ${({ dimension }) => (dimension === 'm' ? 20 : 16)}px;
  font-weight: normal;
  font-family: var(--admiral-font-family, ${(p) => p.theme.fontFamily});
  color: ${({ disabled, theme }) =>
    disabled
      ? `var(--admiral-color-Neutral_Neutral30, ${theme.color['Neutral/Neutral 30']})`
      : `var(--admiral-color-Neutral_Neutral90, ${theme.color['Neutral/Neutral 90']})`};
  user-select: none;
`;

export const Label = styled.label<{ dimension: CheckboxDimension; disabled?: boolean }>`
  display: flex;
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  width: fit-content;
  ${styleTextMixin};
`;
