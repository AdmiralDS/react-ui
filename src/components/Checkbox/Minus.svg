<svg viewBox="-5 -9 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect width="10" height="2" rx="1" fill="white"/>
</svg>
