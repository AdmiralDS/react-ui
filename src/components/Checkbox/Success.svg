<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M13.4714 5.80473C13.7318 5.54438 13.7318 5.12227 13.4714 4.86192C13.2111 4.60157 12.7889 4.60157 12.5286 4.86192L7 10.3905L4.13807 7.52859C3.87772 7.26824 3.45561 7.26824 3.19526 7.52859C2.93491 7.78893 2.93491 8.21105 3.19526 8.47139L6.5286 11.8047C6.78894 12.0651 7.21106 12.0651 7.4714 11.8047L13.4714 5.80473Z"
          fill="white"/>
</svg>
