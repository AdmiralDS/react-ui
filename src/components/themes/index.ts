export * from './light';
export * from './dark';
export type { Theme } from './common';
