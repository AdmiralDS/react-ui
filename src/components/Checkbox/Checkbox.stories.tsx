import React, { useEffect, useState } from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Checkbox } from './index';

import { withDesign } from 'storybook-addon-designs';
import { ALL_DIMENSIONS_VALUES } from '#src/components/Checkbox/CheckboxDimension';
import styled from 'styled-components';
import { T } from '#src/components/T';

export default {
  title: 'Admiral-2.1/Atoms/Checkbox',
  decorators: [withDesign],
  component: Checkbox,
  parameters: {
    design: [
      {
        type: 'figma',
        url: 'https://www.figma.com/file/EGEGZsx8WhdxpmFKu8J41G/Admiral-2.1-UI-Kit?node-id=37%3A20804',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/EGEGZsx8WhdxpmFKu8J41G/Admiral-2.1-UI-Kit?node-id=37%3A21015',
      },
    ],
  },
  argTypes: {
    dimension: {
      options: ALL_DIMENSIONS_VALUES,
      control: { type: 'radio' },
    },
    indeterminate: {
      control: { type: 'boolean' },
    },
    checked: {
      options: [false, true],
      control: { type: 'radio' },
    },
    disabled: {
      control: { type: 'boolean' },
    },
    error: {
      control: { type: 'boolean' },
    },
  },
} as ComponentMeta<typeof Checkbox>;
const Container = styled.div`
  display: flex;
  flex-direction: column;

  > *:not(:first-child) {
    margin-top: 16px;
  }
`;

const Row = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  > *:not(:last-child) {
    margin-right: 66px;
  }
`;

const CheckboxDemo: ComponentStory<typeof Checkbox> = (props) => {
  const args = (Object.keys(props) as Array<keyof typeof props>).reduce((acc, key) => {
    if (props[key] !== undefined) acc[key] = props[key];

    return acc;
  }, {} as Record<any, any>);

  const [checked, setChecked] = useState<boolean>(args.checked ?? false);
  useEffect(() => {
    setChecked(Boolean(args.checked));
  }, [args.checked]);
  return (
    <>
      <Container>
        <T font="Header/H6" as="div">
          Состояния
        </T>
        <Row>
          <Checkbox
            {...args}
            checked={checked}
            onChange={(e) => {
              setChecked(e.target.checked);
            }}
          />
          <Checkbox dimension="s" />
          <T font="Body/Body 1 Long" as="div">
            Default
          </T>
        </Row>
        <Row>
          <Checkbox indeterminate />
          <Checkbox dimension="s" indeterminate />
          <T font="Body/Body 1 Long" as="div">
            Частично выбранный
          </T>
        </Row>
        <Row>
          <Checkbox disabled />
          <Checkbox dimension="s" disabled />
          <T font="Body/Body 1 Long" as="div">
            Disable
          </T>
        </Row>
        <Row>
          <Checkbox disabled defaultChecked />
          <Checkbox dimension="s" disabled defaultChecked />
          <T font="Body/Body 1 Long" as="div">
            Disable active
          </T>
        </Row>
        <Row>
          <Checkbox error />
          <Checkbox dimension="s" error />
          <T font="Body/Body 1 Long" as="div">
            Состояние ошибки
          </T>
        </Row>
      </Container>
    </>
  );
};

export const Example = CheckboxDemo.bind({});
