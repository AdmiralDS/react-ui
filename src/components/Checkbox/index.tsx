import type { InputHTMLAttributes } from 'react';
import * as React from 'react';
import styled, { css } from 'styled-components';
import { ReactComponent as CheckSVG } from './Success.svg';
import { ReactComponent as IndeterminateSVG } from './Minus.svg';
import type { CheckboxDimension } from './CheckboxDimension';

export interface CheckBoxProps extends InputHTMLAttributes<HTMLInputElement> {
  dimension?: CheckboxDimension;
  indeterminate?: boolean;
  error?: boolean;
  children?: never;
}

const width = css<{ dimension: CheckboxDimension }>`
  width: ${({ dimension }) => {
    switch (dimension) {
      case 'm':
        return '20px';
      case 's':
        return '16px';
      default:
        return '20px';
    }
  }};

  fieldset[data-dimension='s'] & {
    width: 16px;
  }
`;

const height = css<{ dimension: CheckboxDimension }>`
  height: ${({ dimension }) => {
    switch (dimension) {
      case 'm':
        return '20px';
      case 's':
        return '16px';
      default:
        return '20px';
    }
  }};
  fieldset[data-dimension='s'] & {
    height: 16px;
  }
`;

const Check = styled(CheckSVG)`
  pointer-events: none;
  & *[fill^='#'] {
    fill: ${({ theme }) => theme.color.text.staticWhite};
  }
`;

const Indeterminate = styled(IndeterminateSVG)`
  pointer-events: none;
  & *[fill^='#'] {
    fill: ${({ theme }) => theme.color.text.staticWhite};
  }
`;

const Container = styled.div<{
  dimension: CheckboxDimension;
}>`
  position: relative;
  box-sizing: border-box;

  overflow: visible;
  ${width};
  ${height};
`;

export const Background = styled.div<{ error?: boolean }>`
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  transition: background 0.1s ease-in;

  /* disable inheritance from parent elements */
  line-height: initial;
  background-color: ${({ theme }) => theme.color.background.primary};
  border: 1px solid ${({ error, theme }) => (error ? theme.color.status.danger : theme.color.text.secondary)};
`;

const Input = styled.input<{ indeterminate?: boolean }>`
  appearance: none;
  ::-ms-check {
    display: none;
  }
  width: calc(100% + 2px);
  height: calc(100% + 2px);

  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;

  :disabled {
    pointer-events: none;
  }

  &:checked + ${Background} {
    background-color: ${({ theme }) => theme.color.basic.primary};
    border: none;
  }

  &:not(:checked) + ${Background} {
    background-color: ${({ theme, indeterminate }) => indeterminate && theme.color.basic.primary};
    border: ${({ indeterminate }) => indeterminate && 'none'};
    > * {
      display: ${(p) => (p.indeterminate ? 'block' : 'none')};
    }
  }

  &:not(:checked):focus + ${Background}, &:not(:disabled):not(:checked):hover + ${Background} {
    background-color: ${({ theme, indeterminate }) =>
      indeterminate ? theme.color.basic.hover : theme.color.background.secondary};
    border: ${(p) => (p.indeterminate ? 'none' : `1px solid ${p.theme.color.basic.hover}`)};
  }

  &:checked:focus + ${Background}, &:not(:disabled):checked:hover + ${Background} {
    background-color: ${({ theme }) => theme.color.basic.hover};
    border: none;
  }

  &:not(:checked):disabled + ${Background} {
    background-color: ${({ theme }) => theme.color.background.primary};
    border: 1px solid ${({ theme }) => theme.color.basic.disable};
  }

  &:disabled + ${Background} {
    background-color: ${({ theme }) => theme.color.basic.disable};
    border: none;
  }

  &:focus-visible {
    outline-offset: 2px;
    outline: ${(p) => p.theme.color.basic.hover} solid 2px;
  }
`;

export const Checkbox = React.forwardRef<HTMLInputElement, CheckBoxProps>(
  ({ className, dimension = 'm', ...props }, ref) => {
    return (
      <Container dimension={dimension} className={className}>
        <Input ref={ref} {...props} type="checkbox" indeterminate={props.indeterminate} />
        <Background error={props.error}>
          {props.indeterminate ? (
            <Indeterminate aria-hidden="true" focusable="false" />
          ) : (
            <Check aria-hidden="true" focusable="false" />
          )}
        </Background>
      </Container>
    );
  },
);

Checkbox.displayName = 'Checkbox';
