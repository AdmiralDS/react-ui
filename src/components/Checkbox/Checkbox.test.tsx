import { LIGHT_THEME } from '#src/components/themes';
import { ThemeProvider } from 'styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Checkbox } from '../Checkbox';

describe('Checkbox', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const checkboxRequiredProps = {
    onChange: jest.fn(),
    checked: false,

    'data-testid': 'element',
  };

  it('should render component with required properties', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} disabled />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small dimension', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} dimension="s" />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange when user clicks on component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} />
      </ThemeProvider>,
    );
    fireEvent.click(wrapper.getByTestId('element'));
    const { onChange } = checkboxRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });

  it('should focus input if user press Tab key', async () => {
    const user = userEvent.setup();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} />
      </ThemeProvider>,
    );
    const [radio] = screen.getAllByTestId('element');
    await user.tab();
    expect(radio).toHaveFocus();
  });

  it('should call onChange if input is focused and user press Space key', async () => {
    const user = userEvent.setup();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Checkbox {...checkboxRequiredProps} />
      </ThemeProvider>,
    );
    const [checkbox] = screen.getAllByTestId('element');
    await user.tab();
    await user.type(checkbox, '{Space}');
    const { onChange } = checkboxRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });

  it('base react checkbox', async () => {
    const user = userEvent.setup();
    render(<input type="checkbox" {...checkboxRequiredProps} />);
    const [checkbox] = screen.getAllByTestId('element');
    await user.tab();
    expect(checkbox).toHaveFocus();
    await user.type(checkbox, '{Space}');
    const { onChange } = checkboxRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });
});
