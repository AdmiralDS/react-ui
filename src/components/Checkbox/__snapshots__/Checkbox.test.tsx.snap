// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render component with disabled state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<body>
    <div>
      <div
        class="c0 c1"
      >
        <input
          class="c2"
          data-testid="element"
          disabled=""
          type="checkbox"
        />
        <div
          class="c3 c4"
        >
          <div
            aria-hidden="true"
            class="c5"
            focusable="false"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<div>
    <div
      class="c0 c1"
    >
      <input
        class="c2"
        data-testid="element"
        disabled=""
        type="checkbox"
      />
      <div
        class="c3 c4"
      >
        <div
          aria-hidden="true"
          class="c5"
          focusable="false"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox should render component with required properties 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<body>
    <div>
      <div
        class="c0 c1"
      >
        <input
          class="c2"
          data-testid="element"
          type="checkbox"
        />
        <div
          class="c3 c4"
        >
          <div
            aria-hidden="true"
            class="c5"
            focusable="false"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<div>
    <div
      class="c0 c1"
    >
      <input
        class="c2"
        data-testid="element"
        type="checkbox"
      />
      <div
        class="c3 c4"
      >
        <div
          aria-hidden="true"
          class="c5"
          focusable="false"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox should render component with small dimension 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 16px;
  height: 16px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<body>
    <div>
      <div
        class="c0 c1"
      >
        <input
          class="c2"
          data-testid="element"
          type="checkbox"
        />
        <div
          class="c3 c4"
        >
          <div
            aria-hidden="true"
            class="c5"
            focusable="false"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  pointer-events: none;
}

.c5 *[fill^='#'] {
  fill: #FFFFFF;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 16px;
  height: 16px;
}

fieldset[data-dimension='s'] .c0 {
  width: 16px;
}

fieldset[data-dimension='s'] .c0 {
  height: 16px;
}

.c4 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c2::-ms-check {
  display: none;
}

.c2:disabled {
  pointer-events: none;
}

.c2:checked + .c3 {
  background-color: #0D69F2;
  border: none;
}

.c2:not(:checked) + .c3 > * {
  display: none;
}

.c2:not(:checked):focus + .c3,
.c2:not(:disabled):not(:checked):hover + .c3 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c2:checked:focus + .c3,
.c2:not(:disabled):checked:hover + .c3 {
  background-color: #0A54C2;
  border: none;
}

.c2:not(:checked):disabled + .c3 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c2:disabled + .c3 {
  background-color: #C4CAD4;
  border: none;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

<div>
    <div
      class="c0 c1"
    >
      <input
        class="c2"
        data-testid="element"
        type="checkbox"
      />
      <div
        class="c3 c4"
      >
        <div
          aria-hidden="true"
          class="c5"
          focusable="false"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
